src/views/portfolio/projects/ProjectDependencyGraph.vue

```diff
@@ -64,78 +64,72 @@ export default {
-      if (this.$route.params.componentUuid) {
-        if (this.project && this.project.directDependencies) {
-          this.$emit('total', 1);
-          this.loading = true
-          let url = `${this.$api.BASE_URL}/${this.$api.URL_COMPONENT}/project/${this.project.uuid}/dependencyGraph/${this.$route.params.componentUuid}`
-          this.axios.get(url).then(response => {
-            if (response.data && Object.keys(response.data).length > 0){
-              this.notFound = false
-              this.response = response
-              this.data = {
-                id: this.nodeId,
-                label: this.createNodeLabel(this.project),
-                objectType: "PROJECT",
-                children: this.transformDependenciesToOrgTreeWithSearchedDependency(this.response.data, {gatheredKeys: []}, !this.showCompleteGraph),
-                fetchedChildren: true,
-                expand: true
-              }
-              this.loading = false
-              new Promise(resolve => setTimeout(resolve, 50)).then(() => {
-                document.getElementsByClassName("searched").item(0).scrollIntoView({
-                  behavior: "smooth",
-                  inline: "center",
-                  block: "center"
-                })
-              })
-            } else {
-              this.$route.query.dependencyGraph = null
-              this.notFound = true
-              this.data = {
-                id: this.nodeId,
-                label: this.createNodeLabel(this.project),
-                objectType: "PROJECT",
-                children: this.transformDependenciesToOrgTree(JSON.parse(this.project.directDependencies), true, {gatheredKeys: []}, this.project.uuid, "PROJECT"),
-                fetchedChildren: true
-              }
-              this.loading = false
-            }
+      // prepare base object
+      const data = {
+        id: this.nodeId,
+        label: this.createNodeLabel(this.project),
+        objectType: "PROJECT"
+      }
+      // do not assign data to this.data yet, otherwise tree breaks :(
+
+      // project has no tree data
+      if (!this.project || !this.project.directDependencies) {
+        this.$emit('total', 0);
+        this.data = data;
+        return;
+      }
+
+      // tree available, populate common info
+      this.$emit('total', 1);
+      data.fetchedChildren = true;
+
+      // full tree, not searching components
+      if(!this.$route.params.componentUuids) {
+        data.children = this.transformDependenciesToOrgTree(JSON.parse(this.project.directDependencies), true, {gatheredKeys: []}, this.project.uuid, "PROJECT");
+        this.data = data;
+        return;
+      }
+
+      // tree with component search active
+      this.createSearchedComponentLookupTable(this.$route.params.componentUuids);
+      this.loading = true
+      let url = `${this.$api.BASE_URL}/${this.$api.URL_COMPONENT}/project/${this.project.uuid}/dependencyGraph/${this.$route.params.componentUuids}`
```

Comment:
It seems like `componentUuids` must be URL encoded. In my testing I am receiving 400s from the API when more than one component is requested. The 400 is returned even before the request reaches the `ComponentResource#getDependencyGraphForComponent` method.

For reference, here is an example URL:

```
http://localhost:8081/api/v1/component/project/d8e78c10-cf81-4733-a206-a0abc296c39a/dependencyGraph/dd65ef43-8c21-4ce9-9ff6-e04cec141eb5|fa3d689e-39af-46f8-b330-294b2da669dd
```

When I encode the pipe (`|`) as `%7C`, I am getting through to the API resource:

```suggestion
      let url = `${this.$api.BASE_URL}/${this.$api.URL_COMPONENT}/project/${this.project.uuid}/dependencyGraph/${encodeURIComponent(this.$route.params.componentUuids)}`
```

Reply:
Interesting, in my case it seems to work without encoding because Chrome does encode it automatically. I end up at an encoded URL without doing it in the code. However, I added this now.

Reply:
Ha, I was low-key expecting the browser to play a role here. I'm using Firefox. So annoying that we still have differing behavior in such basic functionalities across browsers...